ArcGISRuntimeSDKQt_CppSamples/Analysis/FindFeaturesUtilityNetwork/FindFeaturesUtilityNetwork.h

```diff
@@ -0,0 +1,112 @@
+{
+namespace ArcGISRuntime
+{
+class Map;
+class MapQuickView;
+class UtilityNetwork;
+class UtilityElement;
+class UtilityTerminal;
+class UtilityTraceParameters;
+class FeatureLayer;
+class ServiceFeatureTable;
+class SimpleMarkerSymbol;
+class SimpleLineSymbol;
+class GraphicsOverlay;
+class ArcGISFeature;
+class IdentifyLayerResult;
+}
+}
+
+#include <QObject>
+#include <QUrl>
+#include <QUuid>
+#include "Point.h"
+
+class FindFeaturesUtilityNetwork : public QObject
+{
+  Q_OBJECT
+
+  Q_PROPERTY(Esri::ArcGISRuntime::MapQuickView* mapView READ mapView WRITE setMapView NOTIFY mapViewChanged)
+  Q_PROPERTY(bool terminalDialogVisisble MEMBER terminalDialogVisisble NOTIFY terminalDialogVisisbleChanged)
+  Q_PROPERTY(bool dialogVisible MEMBER dialogVisible NOTIFY dialogVisibleChanged)
+  Q_PROPERTY(QString dialogText MEMBER dialogText NOTIFY dialogTextChanged)
+  Q_PROPERTY(bool startingLocationsEnabled MEMBER startingLocationsEnabled NOTIFY startingLocationsChanged)
+  Q_PROPERTY(bool busy MEMBER m_busy NOTIFY busyChanged)
+
+public:
+  explicit FindFeaturesUtilityNetwork(QObject* parent = nullptr);
+  ~FindFeaturesUtilityNetwork();
+
+  static void init();
+
+  Q_INVOKABLE void multiTerminalIndex(int index);
+  Q_INVOKABLE void trace();
+  Q_INVOKABLE void reset();
+
+signals:
+  void mapViewChanged();
+  void terminalDialogVisisbleChanged();
+  void dialogVisibleChanged();
+  void dialogTextChanged();
+  void startingLocationsChanged();
+  void busyChanged();
+
+private slots:
+  void selectFeatures();
+  void selectUtilityElement(QUuid, const QList<Esri::ArcGISRuntime::IdentifyLayerResult*>& results);
+
+private:
+  Esri::ArcGISRuntime::MapQuickView* mapView() const;
+  void setMapView(Esri::ArcGISRuntime::MapQuickView* mapView);
+  void connectSignals();
+  void updateTraceParams(Esri::ArcGISRuntime::UtilityElement* element);
+
+  Esri::ArcGISRuntime::Map* m_map = nullptr;
+  Esri::ArcGISRuntime::MapQuickView* m_mapView = nullptr;
+  Esri::ArcGISRuntime::FeatureLayer* m_deviceLayer = nullptr;
+  Esri::ArcGISRuntime::FeatureLayer* m_lineLayer = nullptr;
+  Esri::ArcGISRuntime::ServiceFeatureTable* m_deviceFeatureTable = nullptr;
+  Esri::ArcGISRuntime::ServiceFeatureTable* m_lineFeatureTable = nullptr;
+  Esri::ArcGISRuntime::SimpleMarkerSymbol* m_startingSymbol = nullptr;
+  Esri::ArcGISRuntime::SimpleMarkerSymbol* m_barrierSymbol = nullptr;
+  Esri::ArcGISRuntime::SimpleLineSymbol* m_lineSymbol = nullptr;
+  Esri::ArcGISRuntime::GraphicsOverlay* m_graphicsOverlay = nullptr;
+  Esri::ArcGISRuntime::UtilityNetwork* m_utilityNetwork = nullptr;
+  Esri::ArcGISRuntime::UtilityTraceParameters* m_traceParams = nullptr;
+  Esri::ArcGISRuntime::ArcGISFeature* m_feature = nullptr;
+
+  const QUrl m_serviceUrl = QUrl("https://sampleserver7.arcgisonline.com/arcgis/rest/services/UtilityNetwork/NapervilleElectric/FeatureServer");
+  bool terminalDialogVisisble = false;
+  bool dialogVisible = false;
+  bool startingLocationsEnabled = true;
```

Comment:
all class members should be prefixed with `m_` for consistency. These and `dialogText` below as well.